Sources/Lexer.swift

```diff
@@ -105,64 +108,49 @@ class Scanner {
     return content.isEmpty
   }
 
-  func scan(until: String, returnUntil: Bool = false) -> String {
-    var index = content.startIndex
-
-    if until.isEmpty {
-      return ""
-    }
-
-    range = range.upperBound..<range.upperBound
-    while index != content.endIndex {
-      let substring = String(content[index...])
-
-      if substring.hasPrefix(until) {
-        let result = String(content[..<index])
+  func scanForTokenEnd(_ tokenChar: Unicode.Scalar) -> String {
+    var foundChar = false
 
-        if returnUntil {
-          range = range.lowerBound..<originalContent.index(range.upperBound, offsetBy: until.count)
-          content = String(substring[until.endIndex...])
-          return result + until
-        }
-
-        content = substring
+    for (index, char) in zip(0..., content.unicodeScalars) {
+      if foundChar && char == Scanner.tokenEndDelimiter {
+        let startIndex = content.index(content.startIndex, offsetBy: index)
+        let result = String(content[..<startIndex])
+        content = String(content[content.index(after: startIndex)...])
+        range = range.upperBound..<originalContent.index(range.upperBound, offsetBy: index + 1)
```

Comment:
why `content` and `range` are needed when `result` is returned right after they are calculated? I can't see them being used.

Reply:
can use `...` instead of `+ 1` ?

Reply:
Yeah, it took me a while before I grokked the whole thing 😄

Reply:
Using `...` gives a `ClosedRange` instead of a `Range`, so we can't use that unfortunately.